skins/cat17/src/app/lib/view_handler/section_info/types/donor_type.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+var objectAssign = require( 'object-assign' ),
+	Base = require( '../base' )
+;
+
+module.exports = objectAssign( Object.create( Base ), {
+	countryNames: null,
+	update: function ( addressType, salutation, title, firstName, lastName, companyName, street, postcode, city, country, email, aggregateValidity ) {
+		var wrapperTag = '<span>',
+			newLongText
+		;
+
+		if ( aggregateValidity.isValid === true ) {
+			this.setSectionStatus( Base.SECTION_STATUS.complete );
+		} else if ( aggregateValidity.isValid === false ) {
+			this.setSectionStatus( Base.SECTION_STATUS.invalid );
+		} else {
+			this.setSectionStatus( Base.SECTION_STATUS.disabled );
+		}
+
+		if ( aggregateValidity.dataEntered ) {
+			this.setIcon( this.getValueIcon( addressType ) );
+		} else {
+			this.setIcon( undefined );
+		}
+
+		if ( this.text ) {
+			this.setText(
+				!aggregateValidity.dataEntered ?
+					this.text.data( Base.DOM_SELECTORS.data.emptyText ) :
+					this.getValueText( addressType )
+			);
+		}
+
+		if ( !this.longText ) {
+			return;
+		}
+
+		newLongText = jQuery( wrapperTag );
+		if ( addressType === 'person' || addressType === 'firma' ) {
```

Comment:
I'm a big fan of either "bailing out early" ( inverting the condition, calling `this.longText.html( newLongText )`and  `setLongTextIndication`, and returning) instead of nesting `if`s too much. Alternatively, you could put the update code in a method.